superchain/superchain.go

```diff
@@ -117,6 +125,23 @@ type PlasmaConfig struct {
 	DAResolveWindow *uint64 `json:"da_resolve_window" toml:"da_resolve_window"`
 }
 
+func (c *ChainConfig) CheckDataAvailability() (DataAvailability, error) {
+	daType := EthDA
+	if c.Plasma != nil {
+		daType = AltDA
+		if c.Plasma.DAChallengeAddress == nil {
+			return "", fmt.Errorf("missing required plasma field: da_challenge_contract_address")
+		}
+		if c.Plasma.DAChallengeWindow == nil {
+			return "", fmt.Errorf("missing required plasma field: da_challenge_window")
+		}
+		if c.Plasma.DAResolveWindow == nil {
+			return "", fmt.Errorf("missing required plasma field: da_resolve_window")
+		}
```

Comment:
I'm not sure how completely you want to validate DA, but checking for the windows seems not needed.  

if `use_plasma` (the deprecated/legacy flag) is specified, the other fields *should also* be specified. Same with `PlasmaConfig`. What that means to me is that the enablement check should only be concerned with that top level. But I am out of my element a bit here, and maybe you do want to do these secondary checks.

Reply:
We're not checking for a `use_plasma` field anymore. We are just checking that `plasma_config` is not nil in the `rollup.json` (e.g. [here](https://github.com/ethereum-optimism/superchain-registry/blob/main/add-chain/testdata/monorepo/op-node/rollup_plasma.json#L34)). If that parent field is present, then we check that the other fields exist. Happy to remove the two window field checks if there is a valid version of the `plasma_config` that does not need them

Reply:
I added checks for the legacy plasma fields in the `rollup.json` file that is an input for the `add-chain` code. The `add-chain` code uses a consistent format (always the nested, dedicated `PlasmaConfig` struct instead of flattened fields) for plasma in its toml config file output